### ISSUE TITLE
uiAreaHandler also need a void *data element

### ISSUE THREAD

**Issue by gnarz** [andlabs/libui#65]
just like the other callbacks.
Rationale: again, scripting integration. Without such an element, the callbacks in the uiAreaHandler will not be able to find the interpreter, unless it is stored in a global variable, which one may not want to do.

**Comment by andlabs** [comment]
I suppose that is doable. The original intent was to let you say this:

```c
struct MyAreaHandler {
    uiAreaHandler firstField;
    myOtherData data;
    myData2 data2;
} s;
```

and use `&s` as a `uiAreaHandler`. (I will say I am using the global state in package ui for Go.)

**Comment by andlabs** [comment]
Merged with #326.
